pkg/lib/cockpit-components-plot.scss

```diff
@@ -33,4 +51,37 @@
         opacity: 0.5;
         shape-rendering: crispEdges;
     }
+
+    &-paths {
+        stroke-width: var(--pf-chart-global--stroke--Width--sm);
+        shape-rendering: geometricPrecision;
+
+        > path {
+            fill: var(--ct-plot-path-color);
+            stroke: var(--ct-plot-path-color);
+        }
+    }
 }
+
+$plotColorCurrent: 0;
+$plotColorTotal: 0;
+
+// Count up total number of plot colors
+@each $plotColor in $plotColors { $plotColorTotal: $plotColorTotal + 1; }
+
+// Iterate through colors and set each graph area to a color
+@each $plotColor, $plotColorBackup in $plotColors {
+    $plotColorCurrent: $plotColorCurrent + 1;
+    .ct-plot-paths > path:nth-child(#{$plotColorTotal}n + #{$plotColorCurrent}) {
```

Comment:
We need to assign colors in reverse, the last `path` element needs to get the first color from the array, the second last `path` gets the second color from the array.  Otherwise all colors change when a new device is hot-plugged while the page is running.

This might or might not be important.  If a new device is is hot-plugged and then you reload the page, the colors change anyway. (But in a different way than they change before reload with this CSS here.)  It's a bit of a mess.

The `path` elements are in reverse order in the DOM because of the how the data structures with the sample are currently constructed. That can of course be changed, but I think in order for the PR to land without undue delay, we shouldn't attempt that here.

Reply:
Oh, there is `:nth-last-child`! I thought we had a real problem here. :-)

Reply:
Done.

Reply:
Wait. nth-last-child does it from reverse... wouldn't it be better to have the elements be in order in the SVG itself instead of doing everything in reverse?